### PR TITLE
[Minor]: Update `architecture.md` to include April tech talks

### PR DESCRIPTION
**Pull request by comphead**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/5499

# Rationale for this change
Add @alamb April architecture tech talks to documentation

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by comphead** [comment]
Putting into draft until part2, part3 of tech talks has released

**Comment by alamb** [comment]
Thank you @comphead

File: docs/source/contributor-guide/architecture.md [hunk 1]
```diff
@@ -21,6 +21,10 @@
...
-- (March 2021): The DataFusion architecture is described in _Query Engine Design and the Rust-Based DataFusion in Apache Arrow_: [recording](https://www.youtube.com/watch?v=K6eCAVEk4kU) (DataFusion content starts [~ 15 minutes in](https://www.youtube.com/watch?v=K6eCAVEk4kU&t=875s)) and [slides](https://www.slideshare.net/influxdata/influxdb-iox-tech-talks-query-engine-design-and-the-rustbased-datafusion-in-apache-arrow-244161934)
-- (February 2021): How DataFusion is used within the Ballista Project is described in \*Ballista: Distributed Compute with Rust and Apache Arrow: [recording](https://www.youtube.com/watch?v=ZZHQaOap9pQ)
+- [Apr 2023]: The Apache Arrow DataFusion Architecture talks series by @alamb
+  - _Query Engine_: [recording](https://youtu.be/NVKujPxwSBA) and [slides](https://docs.google.com/presentation/d/1D3GDVas-8y0sA4c8EOgdCvEjVND4s2E7I6zfs67Y4j8/edit#slide=id.p)
+  - _Logical Plans and Expressions_: [recording](https://youtu.be/EzZTLiSJnhY) and [slides](https://docs.google.com/presentation/d/1ypylM3-w60kVDW7Q6S99AHzvlBgciTdjsAfqNP85K30/edit#slide=id.gbe21b752a6_0_218)
```
Comment (alamb) [inline docs/source/contributor-guide/architecture.md:26]:
```suggestion
  - _Query Engine_: [recording](https://youtu.be/NVKujPxwSBA) and [slides](https://docs.google.com/presentation/d/1D3GDVas-8y0sA4c8EOgdCvEjVND4s2E7I6zfs67Y4j8/edit#slide=id.p)
  - _Logical Plans and Expressions_: [recording](https://youtu.be/EzZTLiSJnhY) and [slides](https://docs.google.com/presentation/d/1ypylM3-w60kVDW7Q6S99AHzvlBgciTdjsAfqNP85K30/edit#slide=id.gbe21b752a6_0_218)
  - _Execution Plans and Execution_:  [recording](https://youtu.be/2jkWU3_w6z0) and [slides](https://docs.google.com/presentation/d/1cA2WQJ2qg6tx6y4Wf8FH2WVSm9JQ5UgmBWATHdik0hg/edit?usp=sharing)
```

**Comment by comphead** [comment]
@alamb please review

**Review by alamb — APPROVED** [review]
LGTM -- thank you very much @comphead

**Pull request merged**